arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -0,0 +1,277 @@
+        }, completion: { [weak self] (result: [AGSSyncLayerResult]?, error: Error?) in
+            SVProgressHUD.dismiss()
+            if let error = error {
+                self?.presentAlert(error: error)
+            } else {
+                self?.geodatabaseDidSync()
+            }
+        })
+    }
+    
+    @IBAction func generateOrSync() {
+        if barButtonItem.title == "Sync geodatabase" {
+            syncGeodatabase()
+        } else {
+            generateGeodatabase()
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Add the source code button item to the right of navigation bar.
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["EditAndSyncFeaturesViewController"]
+    }
+}
+
+// Allows the user to interactively select and move features on the map.
+extension EditAndSyncFeaturesViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Move the selected feature to the tapped location and update it in the feature table.
+        if let feature = self.selectedFeature {
+            let point = mapView.screen(toLocation: screenPoint)
+            if AGSGeometryEngine.geometry(point, intersects: areaOfInterest) {
+                feature.geometry = point
+                feature.featureTable?.update(feature) { [weak self] ( error: Error?) in
+                    if let error = error {
+                        self?.presentAlert(error: error)
+                        self?.selectedFeature = nil
+                        self?.clearSelection()
+                    } else {
+                        self?.didMove()
+                    }
+                }
+            } else {
+                self.presentAlert(title: "Cannot move feature outside downloaded area")
+            }
+        } else { // Identify which feature was tapped and select it.
+            mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 22.0, returnPopupsOnly: false, maximumResultsPerLayer: 1) { [weak self] (results: [AGSIdentifyLayerResult]?, error: Error?) in
+                if let error = error {
+                    self?.presentAlert(error: error)
+                } else if let results = results {
+                    self?.instructionsLabel.text = String("Tap on the map to move the feature")
+                    if let firstResult = results.first {
+                        let layerContent = firstResult.layerContent
+                        
+                        // Check that the result is a feature layer and has elements.
+                        if let featureLayer = layerContent as? AGSFeatureLayer, let feature = firstResult.geoElements.first as? AGSFeature {
+                            featureLayer.select(feature)
+                            self?.selectedFeature = feature
```

Comment:
Having self.selectedFeature both represent application state AND reflect the map's state is a little fragile (and may be behind what @saratchandrakarumuri is seeing).

One option could be to use a `didSet` on `selectedFeature` to ensure that the map's selection matches the `selectedFeature` value/state.

Another would be to make `selectedFeature` a calculated property which works with the map's operationalLayers to determine what's selected, although the unknown there is what to do if more than one thing is selected.

Reply:
I'm not sure what you mean by using a `didSet` on `selectedFeature`. Could you elaborate?

Reply:
See https://github.com/Esri/arcgis-runtime-samples-ios/commit/8ac621fc804690c44a2be69ba6646325dd7cfb05#diff-857df2a38391d55c8b3258543acb5541 for details.

You can execute a `didSet` closure on a variable whenever that variable is updated. In this case I've consolidated all the logic which selects a feature in the map view to the `didSet` on `selectedFeature`, which gives just one entry point to manage both the sample's variable that tracks the selected feature AND the updating of the map view to reflect this.

Reply:
@saratchandrakarumuri if you try with branch [nixta/suggestions-for-pr-750](https://github.com/Esri/arcgis-runtime-samples-ios/tree/nixta/suggestions-for-pr-750) do you see the same behavior?

Reply:
@nixta Your branch fixes the behavior I am seeing. First feature is selected as expected